### PR TITLE
Proddev 261

### PR DESCRIPTION
**Pull request by NightHunterSV**
**Related Issue/Ticket:**

https://openy.atlassian.net/browse/PRODDEV-264
https://openy.atlassian.net/browse/PRODDEV-261

## Steps to test:

- [ ] Log in to VY
- [ ] Ensure the content is ordered initially: videos and blogs -- from the new to old, event entities -- by the event date
- [ ] Ensure you can reorder content using filters and sorts popup (on category and favorites pages, on the listings, etc.)
- [ ] Log in as an admin
- [ ] Go to the VY settings page (/admin/openy/virtual-ymca)
- [ ] Ensure you have the listing ordering settings for each "Component"

image

- [ ] Change the Default view order settings and ensure the changes were applied on VY page reload
- [ ] Ensure filtering is still working for the end user
- [ ] Finally observe the issue is resolved.

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [ ] No updates are necessary for this change.
- [ ] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

File: src/Form/GCSettingsForm.php [hunk 1]
```diff
@@ -146,11 +146,24 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
...
 
-      $form['app_settings']['components'][$id]['empty_block_text'] = [
+      $form['app_settings']['components'][$id]['component']['empty_block_text'] = [
         '#type' => 'textfield',
         '#title' => $this->t('Text for empty block'),
         '#default_value' => $config->get('components.' . $id . '.empty_block_text'),
       ];
+
+      $select_options = [
+        'date_desc' => 'By date (New-Old)',
```
Comment (hamrant) [inline src/Form/GCSettingsForm.php:156]: For events in `date_desc` and `date_asc` we should have different label - `Event Date (desc)` and `Event Date (asc)`

File: js/gated-content/src/views/CategoriesListingPage.vue [hunk 2]
```diff
@@ -75,17 +75,11 @@ export default {
   },
   data() {
     return {
-      DEFAULT_SORT: 'weight_asc',
+      DEFAULT_SORT: 'date_asc',
```
Comment (hamrant) [inline js/gated-content/src/views/CategoriesListingPage.vue:78]: Please add a comment in task with info that `weight_asc` was removed from categories page

**Review by hamrant — CHANGES_REQUESTED** [review]

File: js/gated-content/src/views/CategoriesListingPage.vue (hunk 2 above)
Comment (hamrant) [reply js/gated-content/src/views/CategoriesListingPage.vue:78]: Nevermind

**Review by anpolimus — APPROVED** [review]

**Review by hamrant — APPROVED** [review]

**Comment by AnastasiiaPys** [comment]
tested. testing passed.

**Pull request merged**